src/RuleRightCollection.php

```diff
@@ -249,24 +247,29 @@ public function prepareInputDataForProcess($input, $params)
            //Get all the field to retrieve to be able to process rule matching
             $rule_fields = $this->getFieldsToLookFor();
 
-           //Get all the data we need from ldap to process the rules
-            $sz = @ldap_read(
-                $params_lower["connection"],
-                $params_lower["userdn"],
-                "objectClass=*",
-                $rule_fields
-            );
-            if ($sz === false) {
-                // 32 = LDAP_NO_SUCH_OBJECT => This error can be silented as it just means that search produces no result.
-                if (ldap_errno($params_lower["connection"]) !== 32) {
-                    trigger_error(
-                        AuthLDAP::buildError(
-                            $params_lower["connection"],
-                            sprintf('Unable to get LDAP user having DN `%s` with filter `%s`', $params_lower["userdn"], 'objectClass=*')
-                        ),
-                        E_USER_WARNING
-                    );
+            //Get all the data we need from ldap to process the rules
+            if (!empty($params_lower["connection"])) {
```

Comment:
Maybe it would be more easy to read with:
```php
if (empty($params_lower["connection"])) {
    return rule_parameters;
}

//no if/else needed.
```

Also, I wonder if we can know here we're in the test engine case; in case the contents of `$params_lower` would be wrong outside this usage.

Reply:
I refactored the code.

Looking at what was passed in the parameters, there's nothing to allow us to distinguish normal usage (authentication process) from the rule engine test

Reply:
Hum.... OK; I guess any error in AD configuration should be shown elsewhere.